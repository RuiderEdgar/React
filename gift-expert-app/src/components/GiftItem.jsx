import PropTypes from 'prop-types'
export const GiftItem = ({title, url}) =>  (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
)

//Tarea:
// 1. AÃ±adir proptypes
// 2. evaluar el snapshot

GiftItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}


