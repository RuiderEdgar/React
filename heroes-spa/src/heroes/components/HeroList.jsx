import { getHeroesByPublisher } from "../helpers"
import { HeroCard } from "./HeroCard";

export const HeroList = ({ publisher }) => {
  
  //se puede usar un useMemo para optimizar
  const heroes = getHeroesByPublisher(publisher);

  return (
    <div className="row rows-cols-1 row-cols-md-3">
      {
        heroes.map((heroe) => { 
          return <HeroCard key={heroe.id} {...heroe} />;
        })
      }
    </div>
  )
}

