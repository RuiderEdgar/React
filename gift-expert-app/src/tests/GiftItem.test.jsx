import { render, screen } from "@testing-library/react";
import { GiftItem } from "../components/GiftItem"

describe('Pruebas en <GiftItem/>', ()=>{
    const title = "Pokemon";
    const url = "https://media2.giphy.com/media/fSvqyvXn1M3btN8sDh/giphy.gif?cid=89873216fptjum8d5q4prwg2n7cnqc461pldr3gm7jerfhch&ep=v1_gifs_search&rid=giphy.gif&ct=g";

    test('Debe de evaluar el snapshot', () => { 
        const { container } = render(
          <GiftItem
            title={title}
            url={url}
          />
        );
        expect(container).toMatchSnapshot()
    })

    test('Debe de mostrar la imagen con el /url y el ALT indicado', () => { 
        render(<GiftItem title={title} url={url} />);
        // expect(screen.getByRole("img").src).toBe(url);
        // expect(screen.getByRole("img").alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url)
        expect(alt).toBe(title)
        
    })

    test('Debe de mostrar el titulo en el componente', () => { 
        render(<GiftItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy()

    })
})