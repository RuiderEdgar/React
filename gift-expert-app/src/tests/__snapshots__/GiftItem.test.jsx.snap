// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem/> Debe de evaluar el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pokemon"
      src="https://media2.giphy.com/media/fSvqyvXn1M3btN8sDh/giphy.gif?cid=89873216fptjum8d5q4prwg2n7cnqc461pldr3gm7jerfhch&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Pokemon
    </p>
  </div>
</div>
`;

exports[`Pruebas en GiftItem Debe de evaluar el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pokemon"
      src="https://media2.giphy.com/media/fSvqyvXn1M3btN8sDh/giphy.gif?cid=89873216fptjum8d5q4prwg2n7cnqc461pldr3gm7jerfhch&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Pokemon
    </p>
  </div>
</div>
`;
